fs2/config/__init__.py

```diff
@@ -281,4 +292,22 @@ def load_config_from_path(path: Path) -> "FastSpeech2Config":
             config = FastSpeech2Config(**config)
         return config
 
+    @model_validator(mode="before")
+    @classmethod
+    def check_and_upgrade_checkpoint(cls, data: Any) -> Any:
+        """
+        Check model's compatibility and possibly upgrade.
+        """
+        ckpt_version = data.get("VERSION", "0.0")
+        if ckpt_version > LATEST_VERSION:
```

Comment:
Dang, I just found a bug: 
```
>>> "10.0">"9.0"
False
```
This test needs to do version-style greater than, not string greater than, in all three PRs.